### ISSUE TITLE
OpenXR + OpenGL or Vulkan on Windows

### ISSUE THREAD

**Issue by lostmsu** [maluoi/OpenXRSamples#1]
Hi!

In [the blog post](https://playdeck.net/blog/introduction-to-openxr) you mentioned, that it is possible to use OpenGL with OpenXR. Is it actually possible on Windows? The runtime seems only be supporting DX...

**Comment by maluoi** [comment]
That's right! The implementation here is DirectX specific, but that's largely because it's the graphics API I'm good at, and my focus is on HoloLens which is DX only!

There's a few spots you'll need to make changes:

---

First one is a define up at the top! This activates code in openxr_platform.h:

`#define XR_USE_GRAPHICS_API_D3D11`

You can see some of the other defines [on the spec here](https://www.khronos.org/registry/OpenXR/specs/1.0/html/xrspec.html#platform-specific-header-control), you're probably looking for ` XR_USE_GRAPHICS_API_OPENGL`!

---

`xrCreateInstance` activates extensions, so you'll need to ensure you've got the correct graphics API extension in that list! You can see here I'm using `XR_KHR_D3D11_ENABLE_EXTENSION_NAME `
```C
const char          *extensions[] = { XR_KHR_D3D11_ENABLE_EXTENSION_NAME };
XrInstanceCreateInfo createInfo   = { XR_TYPE_INSTANCE_CREATE_INFO };
createInfo.enabledExtensionCount      = _countof(extensions);
createInfo.enabledExtensionNames      = extensions;
createInfo.applicationInfo.apiVersion = XR_CURRENT_API_VERSION;
strcpy_s(createInfo.applicationInfo.applicationName, 128, app_name);
xrCreateInstance(&createInfo, &xr_instance);
```

This is listed in the spec as a string, the [openGL one is here](https://www.khronos.org/registry/OpenXR/specs/1.0/html/xrspec.html#XR_KHR_opengl_enable), but you can also find the corresponding #define, `XR_KHR_OPENGL_ENABLE_EXTENSION_NAME` listed in openxr_platform.h!

---

The rest is going to be swapchain work! OpenXR takes your swapchain and displays it, so swapchain creation and display is a little dance done between the graphics API and OpenXR. This part is likely going to be the most tricky part, but I'm not up on OpenGL, so I can't give you much in the way of pointers here =/
